sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -939,6 +939,30 @@ def predict(self, X):
 
         return self._label_binarizer.inverse_transform(y_pred)
 
+    def fit(self, X, y):
+        """Fit the model to data matrix X and target(s) y.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape (n_samples, n_features)
+            The input data.
+
+        y : array-like, shape (n_samples,) or (n_samples, n_outputs)
+            The target values (class labels in classification, real numbers in
+            regression).
+
+        Returns
+        -------
+        self : returns a trained MLP model.
+        """
+        if getattr(self, 'classes_', None) is not None:
+            num_classes = len(unique_labels(y))
+            if num_classes != len(self.classes_):
```

Comment:
I'm not entirely certain that we want to exclude the case where a subset of the classes are fit... but it is admittedly a bit weird. Perhaps it's worth checking what other classifiers with `warm_start` do... though they may well just be broken.